File: Controllers/AdminController.cs
using FoodOrderWeb.BLL;
using FoodOrderWeb.Helpers;
using FoodOrderWeb.Models;
using Microsoft.AspNetCore.Mvc;

namespace FoodOrderWeb.Controllers;

public class AdminController : Controller
{
    private readonly FoodService _foods;
    private readonly CategoryService _categories;
    private readonly UserService _users;
    private readonly OrderService _orders;
    private readonly VoucherService _vouchers;
    private readonly SupportService _support;

    public AdminController(FoodService foods, CategoryService categories, UserService users, OrderService orders, VoucherService vouchers, SupportService support)
    {
        _foods = foods;
        _categories = categories;
        _users = users;
        _orders = orders;
        _vouchers = vouchers;
        _support = support;
    }

    public IActionResult Index()
    {
        if (!IsAdmin()) return RedirectToAction("Login", "Account");
        ViewBag.Revenue = _orders.GetRevenue();
        ViewBag.OrderCount = _orders.GetOrders().Count;
        return View();
    }

    public IActionResult Foods()
    {
        if (!IsAdmin()) return RedirectToAction("Login", "Account");
        ViewBag.Categories = _categories.GetAll();
        return View(_foods.GetAll());
    }

    [HttpPost]
    public IActionResult SaveFood(Food food)
    {
        if (!IsAdmin()) return RedirectToAction("Login", "Account");
        _foods.Save(food);
        return RedirectToAction(nameof(Foods));
    }

    [HttpPost]
    public IActionResult DeleteFood(int foodId)
    {
        if (!IsAdmin()) return RedirectToAction("Login", "Account");
        _foods.Delete(foodId);
        return RedirectToAction(nameof(Foods));
    }

    public IActionResult Categories()
    {
        if (!IsAdmin()) return RedirectToAction("Login", "Account");
        return View(_categories.GetAll());
    }

    [HttpPost]
    public IActionResult SaveCategory(Category category)
    {
        if (!IsAdmin()) return RedirectToAction("Login", "Account");
        _categories.Save(category);
        return RedirectToAction(nameof(Categories));
    }

    public IActionResult Users()
    {
        if (!IsAdmin()) return RedirectToAction("Login", "Account");
        return View(_users.GetAll());
    }

    [HttpPost]
    public IActionResult SetUserActive(int userId, bool isActive)
    {
        if (!IsAdmin()) return RedirectToAction("Login", "Account");
        _users.SetActive(userId, isActive);
        return RedirectToAction(nameof(Users));
    }

    public IActionResult Orders()
    {
        if (!IsAdmin()) return RedirectToAction("Login", "Account");
        return View(_orders.GetOrders());
    }

    [HttpPost]
    public IActionResult UpdateOrderStatus(int orderId, string status)
    {
        if (!IsAdmin()) return RedirectToAction("Login", "Account");
        _orders.UpdateStatus(orderId, status);
        return RedirectToAction(nameof(Orders));
    }

    public IActionResult Vouchers()
    {
        if (!IsAdmin()) return RedirectToAction("Login", "Account");
        return View(_vouchers.GetAll());
    }

    [HttpPost]
    public IActionResult SaveVoucher(Voucher voucher)
    {
        if (!IsAdmin()) return RedirectToAction("Login", "Account");
        _vouchers.Save(voucher);
        return RedirectToAction(nameof(Vouchers));
    }

    public IActionResult SupportRequests()
    {
        if (!IsAdmin()) return RedirectToAction("Login", "Account");
        return View(_support.GetAll());
    }

    private bool IsAdmin()
    {
        var user = HttpContext.Session.GetObject<User>("CurrentUser");
        return user?.Role.Equals("Admin", StringComparison.OrdinalIgnoreCase) == true;
    }
}

File: backend-spring/src/main/java/com/foodorder/controller/AdminController.java
package com.foodorder.controller;

import com.foodorder.dto.CategorySaveRequest;
import com.foodorder.dto.DashboardResponse;
import com.foodorder.dto.FoodSaveRequest;
import com.foodorder.dto.OrderStatusRequest;
import com.foodorder.dto.UserActiveRequest;
import com.foodorder.dto.UserRoleRequest;
import com.foodorder.dto.VoucherSaveRequest;
import com.foodorder.model.Category;
import com.foodorder.model.Food;
import com.foodorder.model.Order;
import com.foodorder.model.SupportRequest;
import com.foodorder.model.User;
import com.foodorder.model.Voucher;
import com.foodorder.service.CategoryService;
import com.foodorder.service.FoodService;
import com.foodorder.service.OrderService;
import com.foodorder.service.SupportService;
import com.foodorder.service.UserService;
import com.foodorder.service.VoucherService;
import java.util.List;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PatchMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/admin")
public class AdminController {
    private final FoodService foods;
    private final CategoryService categories;
    private final UserService users;
    private final OrderService orders;
    private final VoucherService vouchers;
    private final SupportService support;

    public AdminController(FoodService foods, CategoryService categories, UserService users, OrderService orders, VoucherService vouchers, SupportService support) {
        this.foods = foods;
        this.categories = categories;
        this.users = users;
        this.orders = orders;
        this.vouchers = vouchers;
        this.support = support;
    }

    @GetMapping("/dashboard")
    public DashboardResponse dashboard() {
        List<Order> allOrders = orders.getOrders(null);
        return new DashboardResponse(orders.getRevenue(), allOrders.size());
    }

    @GetMapping("/foods")
    public List<Food> foods() { return foods.getAllFoods(null, null); }

    @PostMapping("/foods")
    public Food saveFood(@RequestBody FoodSaveRequest request) { return foods.saveFood(request); }

    @DeleteMapping("/foods/{foodId}")
    public void deleteFood(@PathVariable int foodId) { foods.deleteFoodById(foodId); }

    @GetMapping("/categories")
    public List<Category> categories() { return categories.getAllCategories(false); }

    @PostMapping("/categories")
    public Category saveCategory(@RequestBody CategorySaveRequest request) { return categories.saveCategory(request); }

    @GetMapping("/users")
    public List<User> users() { return users.getAllUsers(); }

    @PatchMapping("/users/{userId}/active")
    public User setUserActive(@PathVariable int userId, @RequestBody UserActiveRequest request) {
        return users.updateUserActive(userId, request);
    }

    @PatchMapping("/users/{userId}/role")
    public User setUserRole(@PathVariable int userId, @RequestBody UserRoleRequest request) {
        return users.updateUserRole(userId, request);
    }

    @GetMapping("/orders")
    public List<Order> orders() { return orders.getOrders(null); }

    @PatchMapping("/orders/{orderId}/status")
    public Order updateOrderStatus(@PathVariable int orderId, @RequestBody OrderStatusRequest request) {
        return orders.updateOrderStatus(orderId, request);
    }

    @GetMapping("/vouchers")
    public List<Voucher> vouchers() { return vouchers.getAllVouchers(); }

    @PostMapping("/vouchers")
    public Voucher saveVoucher(@RequestBody VoucherSaveRequest request) { return vouchers.saveVoucher(request); }

    @GetMapping("/support")
    public List<SupportRequest> supportRequests() { return support.getAllSupportRequests(); }
}
